saas/api/users.py

```diff
@@ -103,3 +108,22 @@ class RegisteredAPIView(UserSmartListMixin, RegisteredBaseAPIView):
         }
     """
     serializer_class = UserSerializer
+
+
+class AgreementSignAPIView(GenericAPIView):
+    serializer_class = AgreementSignSerializer
+    slug_url_kwarg = 'agreement'
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        if serializer.data['read_terms']:
+            slug = self.kwargs.get(self.slug_url_kwarg)
+            try:
+                record = Signature.objects.create_signature(
+                    slug, self.request.user)
+            except Agreement.DoesNotExist:
+                raise Http404
+            return Response({'last_signed': record.last_signed})
```

Comment:
The `serializer_class` field is what will be used to generate the API documentation.

So add `last_signed` as read-only field in `AgreementSignSerializer` and return `AgreementSignSerializer().to_representation('last_signed': record.last_signed, 'read_terms': serializer.validated_data['read_terms']})`; otherwise the API doc will be incorrect.